File: EasyTalentos.Dominio/Entity/Candidato.cs
using EasyTalentos.Dominio.Validacoes;



namespace EasyTalentos.Dominio.Entity
{

    public class Candidato
    {

        public int Id { get; set; }

        [Requirido]
        public string Nome { get; set; }

        [Requirido]
        public string Email { get; set; }

        [Requirido]
        public string Telefone { get; set; }

        public string Linkedin { get; set; }

        [Requirido]
        public string Cidade { get; set; }

        [Requirido]
        public string Estado { get; set; }

        public string Portfolio { get; set; }


        public virtual Disponibilidade Disponibilidade { get; set; }


        public virtual Preferencia Preferencia { get; set; }

        [Requirido]
        public double PretencaoSalario { get; set; }


        public virtual Conhecimento Conhecimento { get; set; }

    }
}

File: EasyTalentos.Web/ClientApp/src/app/models/Candidato.ts
import { Conhecimento } from "./Conhecimento";
import { Preferencia } from "./Preferencia";
import { Disponibilidade } from "./Disponibilidade";

export class Candidato {
  id: number;
  email: string = "";
  nome: string = "";
  telefone: string = "";
  linkedin: string = "";
  cidade: string = "";
  estado: string = "";
  portfolio: string = "";
  disponibilidade: Disponibilidade = new Disponibilidade();
  preferencia: Preferencia = new Preferencia();
  pretencaoSalario: number;
  conhecimento: Conhecimento = new Conhecimento();

}

